domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientIT.java

```diff
@@ -643,6 +648,40 @@ public void updateSectionSingleQuote(TestContext context) {
     }));
   }
 
+    @Test
+  public void selectWithTimeoutSuccess(TestContext context) {
+      PostgresClient client = postgresClient();
+      client.getSQLConnection(2000, conn -> {
+        client.selectSingle(conn, "SELECT 1, pg_sleep(1);", context.asyncAssertSuccess());
+      });
+  }
+
+  @Test
+  public void selectWithTimeoutFailure(TestContext context) {
+      PostgresClient client = postgresClient();
+      client.getSQLConnection(1000, conn -> {
+        client.selectSingle(conn, "SELECT 1, pg_sleep(2);", context.asyncAssertFailure());
+      });
+  }
+
+  @Test
+  public void connectionIsRemoved(TestContext context) throws InterruptedException {
+      PostgresClient client = postgresClient();
+      Object mutex = new Object();
```

Comment:
TestContext uses Async returned by context.async() and not synchronized(mutex): https://vertx.io/docs/vertx-unit/java/#_asynchronous_testing